resources/testharness.js

```diff
@@ -483,7 +483,10 @@ policies and contribution forms [3].
     }
 
     function is_service_worker(worker) {
-        return 'ServiceWorker' in self && worker instanceof ServiceWorker;
+        // The worker object may be from another execution context,
+        // so do not use instanceof here.
+        return 'ServiceWorker' in self &&
+            Object.prototype.toString.call(worker) == '[object ServiceWorker]';
```

Comment:
> Do we test instanceof ServiceWorker elsewhere or are we losing test coverage of that with this change?
 
Good thinking. It seems `is_service_worker` is exclusively used to implement `fetch_tests_from_worker` so it wasn't actually used to assert that something is a service worker. We use `instanceof ServiceWorker` directly in state.https.html.

bzbarsky's suggestion sounds good to me.